### DISCUSSION TITLE
Issues with flowCharacteristics and units in OM

### DISCUSSION THREAD

**Discussion by Mavinebula** [OpenModelica/OpenModelica#12723] [General]
Hi!

I'm trying to implement a simple model in Open Modelica and I'm struguling with an error/warning. 
Apparently the problem is related to the pump flow characteristics, where I have set this: [ flowCharacteristic =  Modelica.Fluid.Machines.BaseClasses.PumpCharacteristics.quadraticFlow (V_flow_nominal = {0, 0.25, 0.5}, head_nominal = {100, 60, 0})]
The warnings are these:
[1] 15:23:32 Translation Warning
[Modelica.Fluid.Machines: 328:5-329:76]: [The following equation is INCONSISTENT due to specified unit information: pump.delta_head_init = bombaaa.pump.flowCharacteristic(pump.V_flow_single_init * 1.1, {0.0, 0.25, 0.5}, {100.0, 60.0, 0.0}) - bombaaa.pump.flowCharacteristic(pump.V_flow_single_init, {0.0, 0.25, 0.5}, {100.0, 60.0, 0.0})](omeditmessagesbrowser:///Modelica.Fluid.Machines?lineNumber=328)

[2] 15:23:32 Translation Warning
The units of following sub-expressions need to be equal:
- sub-expression "pump.V_flow_single_init * 1.1" has unit "m"
- sub-expression "bombaaa.pump.flowCharacteristic().V_flow" has unit "m3/s"

[3] 15:23:32 Scripting Notification
Check of bombaaa completed successfully.
Class bombaaa has 343 equation(s) and 343 variable(s).
205 of these are trivial equation(s).

[1] 17:41:58 Translation Error
[Modelica.Fluid.Machines: 2:1-820:13]: [Cannot instantiate Modelica.Fluid.Machines due to class specialization package.](omeditmessagesbrowser:///Modelica.Fluid.Machines?lineNumber=2)

And the graphical representation and its code is the next:
```modelica
model bombaaa
  Modelica.Fluid.Machines.Pump pump(p_b_start = 6e5, redeclare function flowCharacteristic = Modelica.Fluid.Machines.BaseClasses.PumpCharacteristics.quadraticFlow(V_flow_nominal = {0, 0.25, 0.5}, head_nominal = {100, 60, 0}), N_nominal = 1200, checkValve = true, V = 50, redeclare package Medium = Modelica.Media.Water.StandardWater)  annotation(
    Placement(transformation(origin = {-14, -10}, extent = {{-10, -10}, {10, 10}})));
  Modelica.Fluid.Pipes.DynamicPipe pipe(redeclare package Medium = Modelica.Media.Water.StandardWater, length = 100, diameter = 0.3, height_ab = 50)  annotation(
    Placement(transformation(origin = {32, -8}, extent = {{-10, -10}, {10, 10}})));
  Modelica.Fluid.Vessels.ClosedVolume volume(p_start = 1e5, T_start = 293.15, V = 1, nPorts = 1, redeclare package Medium = Modelica.Media.Water.StandardWater)  annotation(
    Placement(transformation(origin = {-74, -14}, extent = {{-10, -10}, {10, 10}}, rotation = 90)));
  Modelica.Fluid.Vessels.OpenTank tank(level(start = 0), height = 1, crossArea = 1, nPorts = 1, redeclare package Medium = Modelica.Media.Water.StandardWater)  annotation(
    Placement(transformation(origin = {76, 0}, extent = {{-20, -20}, {20, 20}})));
  inner Modelica.Fluid.System system annotation(
    Placement(transformation(origin = {-68, -76}, extent = {{-10, -10}, {10, 10}})));
equation
  connect(volume.ports[1], pump.port_a) annotation(
    Line(points = {{-64, -14}, {-24, -14}, {-24, -10}}, color = {0, 127, 255}));
  connect(pump.port_b, pipe.port_a) annotation(
    Line(points = {{-4, -10}, {22, -10}, {22, -8}}, color = {0, 127, 255}));
  connect(pipe.port_b, tank.ports[1]) annotation(
    Line(points = {{42, -8}, {42, -70}, {76, -70}, {76, -20}}, color = {0, 127, 255}));

annotation(
    uses(Modelica(version = "4.0.0")));
end bombaaa;
```
![Captura de pantalla 2024-07-16 174223](https://github.com/user-attachments/assets/e46de630-dfe7-49c6-8841-7ed75ce78558)

I would really appreciate some help, I'm beginer and facing these issues is a boundary for continuing learning.

Thank you!

**Comment by casella** [comment]
I tried you bombaaa model, I get
```
[1] 19:42:29 Translation Error
[Modelica.Fluid.Vessels: 535:7-536:58]: [Parameter volume.portsData[1].diameter has neither value nor start value, and is fixed during initialization (fixed=true).](omeditmessagesbrowser:///Modelica.Fluid.Vessels?lineNumber=535)
```
Pls make sure you have set all required parameters.

**Comment by paulnelsonpe-max** [comment]
I have the same issue: trying to use the 'prescribed pump' and when I enter the flow characteristics as two lists (flow and head, 3 elements each like {2.27, 11.4, 19.1} "m3/H" and {60,56,49} "m" I get the similar error on the flow characteristics. I have tried various edits to the initialization, but have not got it to work. I am suspecting a bug. I can't edit the source (instantiation) code... How do I do that to fix these issues? thanks.
